jobs/haproxy/templates/drain.erb

```diff
@@ -21,12 +22,21 @@ fi
 
 pid="$(cat ${pidfile})"
 
-haproxy_pids=$(pgrep -P $pid -l | grep 'haproxy$' | awk '{print $1}')
+haproxy_wrapper_pid=$(pgrep -P "$pid" haproxy_wrapper)
+<%- if p('ha_proxy.syslog_server') == "stdout" || p('ha_proxy.syslog_server') == "stderr" -%>
+haproxy_master_pid=$(pgrep -P "$haproxy_wrapper_pid" -x haproxy)
+<%- else -%>
+haproxy_master_pid=$(pgrep -P "$pid" -x haproxy)
+<%- end -%>
```

Comment:
`pgrep` finds pids based on name and parent pids. no magic there. I am using it to find the haproxy_wrapper and haproxy master pids.
I don't agree with your suggestion. Why would you want to set the `haproxy_wrapper_pid` only if haproxy logs to stdout? You want it in both cases. Code was taken from `reload` script so I'd like to keep it the same to avoid confusion.